src/ucp/wireup/select.c

```diff
@@ -567,7 +568,8 @@ ucp_wireup_add_lane_desc(const ucp_wireup_select_info_t *select_info,
 static int ucp_wireup_is_lane_proxy(ucp_ep_h ep, ucp_rsc_index_t rsc_index,
                                     uint64_t remote_event_flags)
 {
-    return !ucp_worker_is_tl_p2p(ep->worker, rsc_index) &&
+    return !ucp_worker_is_tl_p2p(ep->worker, rsc_index,
+                                 ucp_ep_has_cm_lane(ep)) &&
```

Comment:
why need to check that it's cm here?

Reply:
we have to have the same logic everywhere, if TL iface has both connection caps (to iface and to ep)  ucp_worker_is_tl_p2p returns true for CM flow and false for all others.

Reply:
the construct "ucp_worker_is_tl_p2p(ep->worker, rsc_index, ucp_ep_has_cm_lane(ep)" should be helper inline function:
ucp_ep_is_connect_p2p()
it's used from other places as well, including address pack